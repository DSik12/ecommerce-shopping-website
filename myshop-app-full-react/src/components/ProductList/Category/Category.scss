@import "../../shared/ScssBreakpoint/Breakpoint.scss";

$Primary-color: #0099e0;
$background-color: white;

@mixin text-decoration($param) {
    text-decoration: $param;
}

.sidebar {
    display: none;
}

.dropright {
    font-size: 10px;
    a {
        @include text-decoration(none);
        font-size: 16px;
    }
}

#filterLaptop, #filterMobile{
    color: $Primary-color;
    @include text-decoration(underline);
    font-size: 13px;
    cursor: pointer;
}

@media (min-width: $tablet-portrait) {
    .sidebar {
        display: block;
        width: 20%;
        box-sizing: border-box;
        height: 100%;
        margin: 10px 10px 0 5px;
        background-color: $background-color;

        ul {
            position: relative;
            list-style-type: none;

            .dropright:hover ul {
                display: initial;
            }

            li a {
                display: flex;
                margin-top: 5px;
                align-items: center;
                font-size: 14px;
                padding: 10px 0 10px 50px;
                color: $Primary-color;
                text-transform: capitalize;
                @include text-decoration(none);
            }

            li a:hover {
                @include text-decoration(underline);
                // text-decoration: underline;
            }

            #logos {
                position: absolute;
                left: 0px;
                img {
                    width: 40px;
                    height: 40px;
                }
            }

            #arrow {
                position: absolute;
                right: 0px;
                font-size: 12px;
            }

            .dropright {
                position: relative;
            }

            ul {
                position: absolute;
                display: none;
                left: 150px;
                width: 90px;
                top: 0px;
                background-color: $background-color;
                border-radius: 5px;
                box-shadow: 2px 3px 3px rgba(0, 0, 0, 0.1);
                li a {
                    text-align: center;
                    padding-left: 0px;
                    padding-top: 3px;
                    padding-bottom: 3px;
                }
            }
        }
        .filter {
            margin-left: 20px;
        }

        .categoriesHeadingWrapper .viewAll a {
            color: $Primary-color;
            font-size: 12px;
            @include text-decoration(none);
            padding-left: 10%;
        }

        .viewAll a:hover {
            text-decoration: underline;
        }
    }

    #filterLaptop, #filterMobile{
        color: $Primary-color;
        @include text-decoration(underline);
        font-size: 13px;
        cursor: pointer;
    }
}

@media (min-width: $laptop) {
    .sidebar {
        ul ul {
            position: absolute;
            display: none;
            left: 200px;
        }
        .categoriesHeadingWrapper .viewAll a {
            padding-left: 25%;
        }
    }

    #filterLaptop, #filterMobile{
        color: $Primary-color;
        @include text-decoration(underline);
        font-size: 13px;
        cursor: pointer;
    }
}
