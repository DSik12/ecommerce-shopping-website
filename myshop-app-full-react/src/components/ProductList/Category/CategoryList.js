import React from 'react';
import { Link } from 'react-router-dom';
import { categories } from '../productListConfig/productListStatic';

const CategoryList = (itemList) => {
    
    return (
        <>
            <li className="dropright">
                <Link to="/products"><span id="logos"><img src={itemList.category} alt="logo" width="32" height="32" /></span>{itemList.spanEle[0]} &amp; {itemList.spanEle[1]}

                    {
                        categories.includes(JSON.stringify(itemList.category)) ?
                            <span>
                                <span id="arrow">
                                    <strong>â€º</strong>
                                </span>
                                <ul>
                                    <li><Link to="/">{itemList.subCategory[0]}</Link></li>
                                    <li><Link to="/">{itemList.subCategory[1]}</Link></li>
                                    <li><Link to="/">{itemList.subCategory[2]}</Link></li>
                                </ul>
                            </span>
                            :
                            <span></span>

                    }
                </Link>
            </li>

        </>
    )
}

export default CategoryList;