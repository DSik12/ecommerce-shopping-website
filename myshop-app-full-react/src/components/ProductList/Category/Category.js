import React from 'react';
import { Link } from 'react-router-dom';
import electronics from '../../../assets/images/electronics.png';

import CategoryList from './CategoryList';
import data from '../productListConfig/productList.json';

import { categoryList } from '../productListConfig/productListStatic';


const Category = (props) => {

    const filterProductsCategory = (category) => {
        const search = window.location.search;
        if (category === 1) {

            if (search) {
                const existingQuery = search.substring(1, search.length);
                if (!existingQuery.includes(data.laptop) && !existingQuery.includes(data.mobile)) {
                    const productApi = `${existingQuery}&${data.mobile}`;
                    props.navigate(data.products, productApi);
                }
                else {
                    const productApi = existingQuery.replace(/q=laptop/, data.mobile);
                    props.navigate(data.products, productApi);
                }
            }
            else {
                const productApi = data.mobile;
                props.navigate(data.products, productApi)
            }

        } else {
            if (search) {
                const existingQuery = search.substring(1, search.length);
                if (!existingQuery.includes(data.mobile) && !existingQuery.includes(data.laptop)) {
                    const productApi = `${existingQuery}&${data.laptop}`;
                    props.navigate(data.products, productApi);
                }
                else {
                    const productApi = existingQuery.replace(/q=Smartphones/, data.laptop);
                    props.navigate(data.products, productApi);
                }
            }
            else {
                props.navigate(data.products, data.laptop);
            }

        }

    };
    const handleFilterLaptop = () => {
        filterProductsCategory(2);
    }
    const handleFilterMobile = () => {
        filterProductsCategory(1);
    }

    const productCategoryList = categoryList.map((category, index) => {
        return <CategoryList category={category.category} subCategory={category.subCategory} spanEle={category.spanEle} key={index} />
    })

    return (
        <>
            <h3>Products</h3>
            <nav>
                <div className="categoriesHeadingWrapper">
                    <span className="categoriesHeading"><strong>Categories</strong> </span>
                    <span className="viewAll">
                        <Link to={data.products0to9} id="viewAll">View All</Link></span>
                </div>
                <hr />
                <ul data-testid="categoryListEl">
                    <li className="dropright">
                        <Link to="/"><span id="logos"><img src={electronics} alt="logo" width="32" height="32" /></span>Electronics &amp; Offices <span id="arrow"><strong>â€º</strong></span></Link>
                        <ul>
                            <li><div id="filterLaptop" onClick={handleFilterLaptop}>Laptops</div></li>
                            <li><div id="filterMobile" onClick={handleFilterMobile}>Mobiles</div></li>
                            <li><div to="/">Office Chair</div></li>
                        </ul>
                    </li>

                    {productCategoryList}

                </ul>
                <hr />
            </nav>
        </>
    )
}

export default Category

