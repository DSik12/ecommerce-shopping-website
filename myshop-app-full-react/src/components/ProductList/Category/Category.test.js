import React from 'react';
import Category from './Category'
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import {render, screen} from '@testing-library/react';
import renderer from 'react-test-renderer'

describe("Category Component", () => {
    it('renders the component properly', () => {
        const divEl = document.createElement('div');
        ReactDom.render(<BrowserRouter><Category/></BrowserRouter>, divEl);
    });

    it('should have products as text', () => {
        render(<BrowserRouter><Category/></BrowserRouter>);
        const productsText = screen.getByText('Products');
        expect(productsText).toBeInTheDocument();
    });

    it('should have categories as text', () => {
        render(<BrowserRouter><Category/></BrowserRouter>);
        const categoriesText = screen.getByText('Categories');
        expect(categoriesText).toBeInTheDocument();
    });

    it('should have view all as text', () => {
        render(<BrowserRouter><Category/></BrowserRouter>);
        const viewAllText = screen.getByText('View All');
        expect(viewAllText).toBeInTheDocument();
    });

    it('should display categories correctly', () => {
        render(<BrowserRouter><Category/></BrowserRouter>);
        const categoryList = screen.getByTestId('categoryListEl');
        expect(categoryList).toHaveTextContent('Electronics & Offices â€ºLaptopsMobilesOffice ChairClothing,Shoes & AccessoriesHome, Furniture & AppliancesToys & Video GameMovies Music & BooksG & aFood,Household & PetsPharmacy & HealthcareBeauty & JwelerySports,Fitness & OutdoorsAuto,Tyres & Industrial');
    });

    it('should have right Category comp snapshot', () => {
        const tree = renderer.create(<BrowserRouter><Category/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
})