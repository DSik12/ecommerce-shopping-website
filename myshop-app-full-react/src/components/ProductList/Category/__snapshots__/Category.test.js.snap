// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component should have right Category comp snapshot 1`] = `
Array [
  <h3>
    Products
  </h3>,
  <nav>
    <div
      className="categoriesHeadingWrapper"
    >
      <span
        className="categoriesHeading"
      >
        <strong>
          Categories
        </strong>
         
      </span>
      <span
        className="viewAll"
      >
        <a
          href="/products?_start=0&_end=9"
          id="viewAll"
          onClick={[Function]}
        >
          View All
        </a>
      </span>
    </div>
    <hr />
    <ul
      data-testid="categoryListEl"
    >
      <li
        className="dropright"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="electronics.png"
              width="32"
            />
          </span>
          Electronics & Offices 
          <span
            id="arrow"
          >
            <strong>
              â€º
            </strong>
          </span>
        </a>
        <ul>
          <li>
            <div
              id="filterLaptop"
              onClick={[Function]}
            >
              Laptops
            </div>
          </li>
          <li>
            <div
              id="filterMobile"
              onClick={[Function]}
            >
              Mobiles
            </div>
          </li>
          <li>
            <div
              to="/"
            >
              Office Chair
            </div>
          </li>
        </ul>
      </li>
      <li
        className="dropright"
      >
        <a
          href="/products"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="clothes.png"
              width="32"
            />
          </span>
          Clothing,Shoes
           & 
          Accessories
          <span />
        </a>
      </li>
      <li
        className="dropright"
      >
        <a
          href="/products"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="furniture.png"
              width="32"
            />
          </span>
          Home, Furniture
           & 
          Appliances
          <span />
        </a>
      </li>
      <li
        className="dropright"
      >
        <a
          href="/products"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="game.png"
              width="32"
            />
          </span>
          Toys
           & 
          Video Game
          <span />
        </a>
      </li>
      <li
        className="dropright"
      >
        <a
          href="/products"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="music.png"
              width="32"
            />
          </span>
          Movies Music
           & 
          Books
          <span />
        </a>
      </li>
      <li
        className="dropright"
      >
        <a
          href="/products"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="garden.png"
              width="32"
            />
          </span>
          G
           & 
          a
          <span />
        </a>
      </li>
      <li
        className="dropright"
      >
        <a
          href="/products"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="food.png"
              width="32"
            />
          </span>
          Food,Household
           & 
          Pets
          <span />
        </a>
      </li>
      <li
        className="dropright"
      >
        <a
          href="/products"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="health.png"
              width="32"
            />
          </span>
          Pharmacy
           & 
          Healthcare
          <span />
        </a>
      </li>
      <li
        className="dropright"
      >
        <a
          href="/products"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="beauty.png"
              width="32"
            />
          </span>
          Beauty
           & 
          Jwelery
          <span />
        </a>
      </li>
      <li
        className="dropright"
      >
        <a
          href="/products"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="sports.png"
              width="32"
            />
          </span>
          Sports,Fitness
           & 
          Outdoors
          <span />
        </a>
      </li>
      <li
        className="dropright"
      >
        <a
          href="/products"
          onClick={[Function]}
        >
          <span
            id="logos"
          >
            <img
              alt="logo"
              height="32"
              src="auto.png"
              width="32"
            />
          </span>
          Auto,Tyres
           & 
          Industrial
          <span />
        </a>
      </li>
    </ul>
    <hr />
  </nav>,
]
`;
